test/powershell/Modules/Microsoft.PowerShell.Utility/FormatOutOfBandData.Tests.ps1

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe 'Formatting out of band data tests' -Tags 'CI' {
+
+    Context 'Outputting out of band data renders properly' {
+        
+        BeforeAll {
+            $formatCommandTestCases = @(
+                @{
+                    FormatCommand = 'Format-Table'
+                    DataType      = 'TableRowEntry'
+                },
+                @{
+                    FormatCommand = 'Format-List'
+                    DataType      = 'ListViewEntry'
+                },
+                @{
+                    FormatCommand = 'Format-Wide'
+                    DataType      = 'WideViewEntry'
+                },
+                @{
+                    FormatCommand = 'Format-Custom'
+                    DataType      = 'ComplexViewEntry'
+                }
+            )
+            $oobTypes = 'Error', 'Warning', 'Verbose', 'Debug', 'Information'
+            $formatCommandWithOobTypeTestCases = foreach ($oobType in $oobTypes) {
+                foreach ($item in $formatCommandTestCases) {
+                    @{
+                        OobType       = $oobType
+                        FormatCommand = $item.FormatCommand
+                        DataType      = $item.DataType
+                    }
+                }
+            }
+
+            function Test-OutOfBandFormatting {
+                [CmdletBinding()]
+                [OutputType([System.Diagnostics.Process])]
+                param(
+                    [Parameter(Mandatory)]
+                    [ValidateSet('Error', 'Warning', 'Verbose', 'Debug', 'Information', 'Number', 'String')]
+                    [string]
+                    $OobType
+                )
+                $ErrorActionPreference = $WarningPreference = $VerbosePreference = $DebugPreference = $InformationPreference = [System.Management.Automation.ActionPreference]::Continue
+                $process = Get-Process -Id $PID
+
+                $process
```

Comment:
If we change default formatting for the type tests will fail? Can we implement more reliable tests? Maybe use a test custom type?

Reply:
The function is always piped into specific format commands, so we're not relying on default formatting here at all. Plus, these tests actually ignore the formatting of the process objects, and focus exclusively on validating the formatting of the out of band data for each of the specific types that use out of band formatting. I don't think these tests are at risk of failing unless out of band formatting handling is changed, in which case we want them to fail because that would indicate a breaking change or, if the change is approved, a test that needs to be updated.